algosdk/abi/reference.py

```diff
@@ -0,0 +1,17 @@
+class ABIReferenceType:
+    # Account reference type
+    ACCOUNT = "account"
+
+    # Application reference type
+    APPLICATION = "application"
+
+    # Asset reference type
+    ASSET = "asset"
+
+
+def is_abi_reference_type(t: "ABIReferenceType") -> bool:
+    return t in (
+        ABIReferenceType.ACCOUNT,
+        ABIReferenceType.APPLICATION,
+        ABIReferenceType.ASSET,
+    )
```

Comment:
nit: I think my suggestion on this one is kinda optional: https://docs.python.org/3/library/enum.html
It feels to me that you can take advantage of `Enum` class, and see if an element is abi_reference type can use a simple `in` operation.
`is_abi_reference_type` can actually accept  any variable type, or according to the suggestion of type hint, the boolean function is vacuous true.

Reply:
That's a good point, I think the type hint should be changed to `Any`. I think Python's regular `Enum` has some issues with string comparison so I opted for this comparison.